src/hotspot/cpu/riscv/relocInfo_riscv.cpp

```diff
@@ -72,13 +72,12 @@ void Relocation::pd_set_data_value(address x, bool verify_only) {
 }
 
 address Relocation::pd_call_destination(address orig_addr) {
-  assert(is_call(), "should be an address instruction here");
-  if (NativeCall::is_at(addr())) {
-    return nativeCall_at(addr())->reloc_destination(orig_addr);
-  }
-  // Non call reloc
-  if (orig_addr != nullptr) {
-    // the extracted address from the instructions in address orig_addr
+  assert(is_call(), "should be a call here");
+  if (orig_addr == nullptr) {
```

Comment:
IIUC, it is synchronized with [JDK-8321509](https://bugs.openjdk.org/browse/JDK-8321509)? Should we add `USE_TRAMPOLINE_STUB_FIX_OWNER` and the other stuff?

Reply:
[JDK-8343430](https://bugs.openjdk.org/browse/JDK-8343430) already removed the trampoline call for RISCV, so we are good, right?

Reply:
Correct,  it is partially synchronized with [JDK-8321509](https://bugs.openjdk.org/browse/JDK-8321509), mainly 4 and 6 [here](https://github.com/openjdk/jdk/pull/19796#issuecomment-2188094970). 
We do not really need `USE_TRAMPOLINE_STUB_FIX_OWNER` after [JDK-8343430](https://bugs.openjdk.org/browse/JDK-8343430).

Reply:
>  it is partially synchronized with [JDK-8321509](https://bugs.openjdk.org/browse/JDK-8321509), mainly 4 and 6 [here](https://github.com/openjdk/jdk/pull/19796#issuecomment-2188094970). 
As you mentioned in PR, this is just cleanup code. Do you think we should add this code in a separate PR?

Reply:
I agree with @feilongjiang that we should keep this cleanup change which shouldn't modify the original logic.

Reply:
Thanks all for the review! 
I will only keep the clean up part and revert the rest of the changes.